### DISCUSSION TITLE
Requirement step

### DISCUSSION THREAD

**Discussion by lilyMrt** [shipshapecode/shepherd#2369] [Ideas]
It would be great if Shepherd could support steps with requirements, which can be dynamically evaluated to display the appropriate step. Here is an example use case:

Step 1: Expand dropdown menu (requirement: menu is collapsed)
Step 1.5: Select menu option (requirement: menu is expanded)

If the state of the menu changes while on either step, it will trigger a change in the tour, going backward of forward as required.

Currently, this cannot be achieved with just advanceOn or showOn without external logic because the conditions are not re-evaluated, and there is no relationship between the two steps to make the tour go backward if the condition is no longer met.

**Comment by RobbieTheWagner** [comment]
@lilyMrt I think this is out of scope, but is definitely doable if you take control of the tour yourself using `tour.next` instead of relying on `advanceOn`.
